tests/core/notification/test_core_event_consumer.py

```diff
@@ -0,0 +1,112 @@
+        self.event_operation_collected[event.operation] = self.event_operation_collected.get(event.operation, 0) + 1
+
+
+class ScenarioCoreEventConsumerProcessor(CoreEventConsumerBase):
+    def __init__(self, registration_id: str, queue: SimpleQueue):
+        self.scenario_event_collected = 0
+        self.event_operation_collected: dict = {}
+        super().__init__(registration_id, queue)
+
+    def process_event(self, event: Event):
+        self.scenario_event_collected += 1
+        self.event_operation_collected[event.operation] = self.event_operation_collected.get(event.operation, 0) + 1
+
+
+class PipelineCreationCoreEventConsumerProcessor(CoreEventConsumerBase):
+    def __init__(self, registration_id: str, queue: SimpleQueue):
+        self.pipeline_event_collected = 0
+        self.creation_event_operation_collected = 0
+        super().__init__(registration_id, queue)
+
+    def process_event(self, event: Event):
+        self.pipeline_event_collected += 1
+        self.creation_event_operation_collected += 1
+
+
+def test_core_event_consumer():
+    register_id_0, register_queue_0 = Notifier.register()
+    event_processor_0 = AllCoreEventConsumerProcessor(register_id_0, register_queue_0)
+
+    register_id_1, register_queue_1 = Notifier.register(entity_type=EventEntityType.SCENARIO)
+    event_processor_1 = ScenarioCoreEventConsumerProcessor(register_id_1, register_queue_1)
+
+    register_id_2, register_queue_2 = Notifier.register(
+        entity_type=EventEntityType.PIPELINE, operation=EventOperation.CREATION
+    )
+    event_processor_2 = PipelineCreationCoreEventConsumerProcessor(register_id_2, register_queue_2)
+
+    event_processor_0.start()
+    event_processor_1.start()
+    event_processor_2.start()
+
+    Config.configure_global_app(clean_entities_enabled=True)
+    dn_config = Config.configure_data_node("dn_config")
+    task_config = Config.configure_task("task_config", print, [dn_config])
+    pipeline_config = Config.configure_pipeline("pipeline_config", [task_config])
+    scenario_config = Config.configure_scenario("scenario_config", [pipeline_config], frequency=Frequency.DAILY)
+
+    scenario = tp.create_scenario(scenario_config)
+    sleep(3)
+
+    # assert_true_after_time(lambda: len(event_processor_0.event_type_collected) == 5, time=10)
+    # assert_true_after_time(lambda: event_processor_0.event_operation_collected[EventOperation.CREATION] == 5, time=10)
+    # assert_true_after_time(lambda: event_processor_1.scenario_event_collected == 1, time=10)
+    # assert_true_after_time(lambda: event_processor_1.event_operation_collected[EventOperation.CREATION] == 1, time=10)
+    # assert_true_after_time(lambda: len(event_processor_1.event_operation_collected) == 1, time=10)
+    # assert_true_after_time(lambda: event_processor_2.pipeline_event_collected == 1, time=10)
+    # assert_true_after_time(lambda: event_processor_2.creation_event_operation_collected == 1, time=10)
```

Comment:
Why don't you use the `assert_true_after_time` at the end? I believe it's cleaner than a `sleep(3)`.

Reply:
I tried as you can see, but it caused failed test in windows and macos github action 😢 It's still causing them :)